test/integration/dialects/yugabytedb/associations.test.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+const chai = require('chai');
+
+const expect = chai.expect;
+const Support = require('../../support');
+
+const dialect = Support.getTestDialect();
+const DataTypes = require('@sequelize/core/lib/data-types');
+
+if (dialect.startsWith('yugabytedb')) {
+  describe('[YUGABYTEDB Specific] associations', () => {
+    describe('many-to-many', () => {
+      describe('where tables have the same prefix', () => {
+        it('should create a table wp_table1wp_table2s', function () {
```

Comment:
> shall we make the postgres's test run for yugabytedb as well there in postgres itself?

I think so :)
It's what we should have done for MariaDB / MySQL to avoid duplicating tests. We can separate tests once they become specific to YugabyteDB